### ISSUE TITLE
Add More Loss Functions in Shifu NN

### ISSUE THREAD

**Issue by zhangpengshan** [ShifuML/shifu#442]
Current NN in Shifu only support loss function is RMS, LogLoss should also be supported. Others like absolute loss can be investigated

**Comment by zhangpengshan** [comment]
https://github.com/encog/encog-java-core/blob/master/src/main/java/org/encog/neural/error/CrossEntropyErrorFunction.java

https://github.com/encog/encog-java-core/blob/master/src/main/java/org/encog/mathutil/error/ErrorCalculation.java

**Comment by zhangpengshan** [comment]
add new Loss in train#params for Neural Network support, by default it is ‘squared’， ‘log’ and 'absolute' can also be evaluated. In NN, log and squared loss have been tested.

**Comment by zhangpengshan** [comment]
Updating doc please
